accepted/00000-rbac.md

```diff
@@ -0,0 +1,285 @@
+
+Access control operates on three distinct levels, each with specific goals and constraints:
+
+1. **Authentication:** Identifies the user calling this action.
+2. **Authorization:** Determines if the user is allowed to call the endpoints for this action.
+3. **Accounting:** Tracks and logs actions taken by users on specific resources.
+
+This proposal specifically addresses level 2, authorization. However, managing access control across different instances or sets of the same resource type is considered beyond the scope of this RFC.
+
+
+# Detailed design
+[design]: #detailed-design
+
+## Overview
+
+This solution stores all available system endpoints in the database and determines which endpoints each user can access, except the **Uyuni Administrator** role, which bypasses access control.
+
+Endpoints are organized into functional groups called **namespaces**, allowing administrators to modify access control rules easily without needing to navigate the complex endpoint structure.
+This approach reduces the risk of misconfiguration.
+
+The namespaces are further categorized by access mode into "View" and "Modify" to provide a clear distinction of access levels.
+Access policies are defined by mapping namespaces to access control groups.
+
+All existing endpoints must be mapped and grouped into namespaces:
+
+- **308** endpoints in Spark framework used by the web UI
+- **757** XML-RPC and JSON over HTTP API endpoints (excluding overloads)
+- **563** Struts entry points
+
+The proposed solution is based on the following key components:
+
+1. Access control data
+2. User access information
+3. Access control filters
+4. Management interface
+5. Development resources
+
+## 1. Access Control Data
+
+Every access point in Uyuni is mapped as an **endpoint** in the database and grouped into **namespaces**.
+Access control is enforced by defining rules for these namespaces, either per **user** or **access group**.
+
+Below is a simplified ER diagram illustrating the proposed structure for access information in the database:
+
+![ER diagram](images/00000-rbac-er-diagram.png)
+*[View in dbdiagram.io](https://dbdiagram.io/d/Uyuni-RBAC-672e18b3e9daa85acacbddd0)*
+
+### Endpoint
+
+An **endpoint** represents a specific access point that can be called by a client, such as a web page URL, an internal API endpoint used by a web page, or a public API endpoint.
+
+- **`endpoint`**: The accessible URI of the endpoint.
+
+- **`class_method`**: The Java class that handles incoming requests to the endpoint, such as a controller class for web endpoints or an API handler for API endpoints.
+
+- **`http_method`**: The HTTP method (e.g., GET, POST, PUT, DELETE) accepted by the endpoint. If multiple methods are supported, each should be defined as a separate endpoint since they typically serve different purposes.
+
+- **`auth_required`**: Indicates if the endpoint requires authorization.
```

Comment:
If endpoint, will we have a separate namespace for such endpoints? What endpoints would usually fall in this category?